pkg/pkgupdates/pkgupdates.jsx

```diff
@@ -0,0 +1,448 @@
+    9: _('Downloading'),
+    10: _('Updating'),
+    11: _('Cleaning up'),
+    14: _('Verifying'),
+}
+
+var dbus_pk = cockpit.dbus('org.freedesktop.PackageKit', {superuser: 'try'});
+
+function pkTransaction() {
+    var dfd = cockpit.defer();
+
+    dbus_pk.call('/org/freedesktop/PackageKit', 'org.freedesktop.PackageKit', 'CreateTransaction', [], {timeout: 5000})
+        .done(result => {
+            var transProxy = dbus_pk.proxy('org.freedesktop.PackageKit.Transaction', result[0]);
+            transProxy.wait(() => {dfd.resolve(transProxy)});
+        })
+        .fail(ex => {dfd.reject(ex)});
+
+    return dfd.promise();
+}
+
+// parse CVEs from an arbitrary text (changelog) and return URL array
+function parseCVEs(text) {
+    if (!text)
+        return [];
+
+    var cves = text.match(/CVE-\d{4}-\d+/g);
+    if (!cves)
+        return [];
+    return cves.map(n => 'https://cve.mitre.org/cgi-bin/cvename.cgi?name=' + n);
+}
+
+function deduplicate(list) {
+    var d = { };
+    list.forEach(i => {if (i) d[i] = true});
+    var result = Object.keys(d);
+    result.sort();
+    return result;
+}
+
+function commaJoin(list) {
+    return list.reduce((prev, cur) => [prev, ', ', cur])
+}
+
+function HeaderBar(props) {
+    var num_updates = Object.keys(props.updates).length;
+    var num_security = 0;
+    var state;
+    if (props.state == 'available') {
+        state = cockpit.ngettext('$0 update', '$0 updates', num_updates);
+        for (var u in props.updates)
+            if (props.updates[u].security)
+                ++num_security;
+        if (num_security > 0)
+            state += cockpit.ngettext(', including $1 security fix', ', including $1 security fixes',  num_security);
+        state = cockpit.format(state, num_updates, num_security);
+    } else
+        state = STATE_HEADINGS[props.state];
+
+    return (
+        <div className='content-header-extra'>
+            <table width='100%'>
+                <tr>
+                    <td id='state'>{state}</td>
+                    {props.state == 'uptodate' || props.state == 'available' ?
+                        <td className='text-right'>{props.timeSinceRefresh ?
+                            <span style={{paddingRight: '3ex'}}>
+                                {cockpit.format(_('Last checked: $0 ago'), moment.duration(props.timeSinceRefresh * 1000).humanize())}
+                            </span>
+                            : null}
+                            <button className='btn btn-default' onClick={() => props.onRefresh()} >Check for updates</button>
+                        </td>
+                        : null}
+                </tr>
+            </table>
+        </div>
+    );
+}
+
+class CVELink extends React.Component {
+    constructor() {
+        super();
+        this.state = {summary: _('No information available')};
+    }
+
+    componentDidMount() {
+        cockpit.http(80, {address: 'cve.circl.lu'}).get('/api/cve/' + this.props.cve)
```

Comment:
This will make a HTTP request from the cockpit-bridge process.  I think you want to use $.get here or similar, so that the browser makes that request.